radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,288 @@
+ */
+static uint32_t PdbVoltageCommandBytes[4] = {
+		0x10E280,		// Output Voltage of BCR
+		0x10E220,		// Output Voltage of Battery Bus
+		0x10E210,		// Output Voltage of 5V Bus
+		0x10E200		// Output Voltage of 3.3V Bus
+};
+
+/**
+ * 4 Byte commands for requesting output current readings for each bus and BCR
+ */
+static uint32_t PdbCurrentCommandBytes[4] = {
+		0x10E284,		// Output Current of BCR in mA
+		0x10E224,		// Output Current of Battery Bus
+		0x10E214,		// Output Current of 5V Bus
+		0x10E204		// Output Current of 3.3V Bus
+};
+
+/**
+ * Global variable for detecting when there is an error in PdbI2cTalk()
+ */
+uint8_t i2cErrorFlag = 0;
+
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static uint16_t PdbI2cTalk(uint32_t command);
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+/**
+ * Get the Irradiance (W/m^2) value from the sun sensor on each face of the CubeSat
+ *
+ * @param A pointer to a SunSensorStatus object, which will be filled with the converted ADC data for each sun sensor
+ * @return an error code, or 0 if no error
+ */
+int getSunSensorData(SunSensorStatus* sunData) {
+
+	// Create a temporary variable for receiving I2C data
+	uint16_t i2c_received;
+
+	// Create a temporary array for calculated sun data before transferring into the structure sunData
+	float convertedData[NUM_SUN_SENSORS];
+
+	// Send 6 commands to get each of the sun sensor's data
+	for(int i = 0; i < NUM_SUN_SENSORS; i = i + 1){
+		i2c_received = PdbI2cTalk(PdbSunSensorCommandBytes[i]);
+		if(i2cErrorFlag == 1){
+			// Do some error raising here?
+			// return the error
+			return i2c_received;
+		}
+		convertedData[i] = ADC_COUNT_TO_IRRADIANCE * i2c_received;
+	}
+
+	// Now store all of the calculated data into the proper slot in the sunData structure
+	sunData->xPos = convertedData[3];
+	sunData->xNeg = convertedData[2];
+	sunData->yPos = convertedData[0];
+	sunData->yNeg = convertedData[1];
+	sunData->zPos = convertedData[4];
+	sunData->zNeg = convertedData[5];
+	// Could change these numbers into an ENUM with the names of the connections
+
+	return 0;
+}
+
+
+/**
+ * Request and store all of the relevant data (voltage, current, temperature, sun sensor data)
+ *  	from the PDB board
+ *
+ * @param A pointer to an PDBStatus object containing all of the telemetry for the PDB system
+ * @return either 0 or an error code
+ */
+int getPdbTelemetry(PdbStatus* dataStorage) {
+
+	// Create an empty sunSensorData structure
+	SunSensorStatus sunSensorData;
```

Comment:
Might want to initialize this with `SunSensorStatus sunSensorData = { 0 };` which will put a zero to ensure that it's not random data from whatever used to be at these memory addresses.